src/pkg/fault/example_fault_test.go

```diff
@@ -67,23 +67,27 @@ func Example_new() {
 func Example_errors_Fail() {
 	errs := fault.New(false)
 
-	// Fail() should be used to record any error that highlights a
+	// Fail() is used to record any error that highlights a
 	// non-recoverable failure in a process.
 	//
 	// Fail() should only get called in the last step before returning
 	// a fault.Errors from a controller.  In all other cases, you
 	// should simply return an error and expect the upstream controller
 	// to call Fail() for you.
-	if err := connectClient(); err != nil {
-		// normally, you'd want to
-		// return errs.Fail(err)
-		errs.Fail(err)
+	topLevelHandler := func(errs *fault.Errors) *fault.Errors {
+		if err := connectClient(); err != nil {
+			return errs.Fail(err)
+		}
+
+		return errs
```

Comment:
What is the difference between returning 
`errs.Fail(err) ` and returning errs... Because either way. there seems to be a need to return errs

Reply:
Fail(err) returns the Errors struct, as a convenience.

`func (e *Errors) Fail(err error) *Errors {`